### PR TITLE
fix: format currency hex values on AMM account pages

### LINKED ISSUE #717: Format currency hex values on AMM account pages
The currency codes shown for balances need to take into account hex based currencies.  Using the `<Currency>` component should fix it.

https://amm-devnet.xrpl.org/accounts/rpwtoDgigJj2rEfbWDghiLXHdm1zSwHaBD

Monosnap XRPL Explorer | rpwtoDgigJj2  2023-05-26 17-32-47

### PR DESCRIPTION
**Pull request by tarunsamanta2k20**
## High Level Overview of Change
The currency codes shown for balances need to take into account hex based currencies. Using the <Currency> component should fix it.

https://amm-devnet.xrpl.org/accounts/rpwtoDgigJj2rEfbWDghiLXHdm1zSwHaBD

Monosnap XRPL Explorer | rpwtoDgigJj2  2023-05-26 17-32-47

### Context of Change

Fixes #717 

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

## Before / After
Before SS:
image

After SS: 
image

### CODE REVIEW

**Comment by ckniffen** [comment]
@tarunsamanta2k20 Looks great! Can you add a before and after image to the PR?

**Comment by tarunsamanta2k20** [comment]
@ckniffen i couldn't view that page because I don't have any accident to use in local. If you have any send me or tell me how to test?

**Comment by ckniffen** [comment]
> @ckniffen i couldn't view that page because I don't have any account to use in local. If you have any send me or tell me how to test?

You can use these environment variables to access the amm devnet locally.

```
VITE_RIPPLED_HOST=amm.devnet.rippletest.net
VITE_P2P_RIPPLED_HOST=ammclio.devnet.rippletest.net
VITE_VALIDATOR=vlamm.devnet.rippletest.net
VITE_ENVIRONMENT=amm
```

**Comment by tarunsamanta2k20** [comment]
> > @ckniffen i couldn't view that page because I don't have any account to use in local. If you have any send me or tell me how to test?
> 
> You can use these environment variables to access the amm devnet locally.
> 
> ```
> VITE_RIPPLED_HOST=amm.devnet.rippletest.net
> VITE_P2P_RIPPLED_HOST=ammclio.devnet.rippletest.net
> VITE_VALIDATOR=vlamm.devnet.rippletest.net
> VITE_ENVIRONMENT=amm
> ```

@ckniffen where do I have to add those?

**Comment by tarunsamanta2k20** [comment]
@ckniffen is it looking good?
image

**Comment by tarunsamanta2k20** [comment]
image

**Comment by ckniffen** [comment]
It should show "AMM2 20" and "AMM1 22". "414D4D3200000000000000000000000000000000" is the hex representation of "AMM2"

**Comment by mvadari** [comment]
You should actually use `Amount`, not `Currency` (with the non-localized balance).

**Comment by mvadari** [comment]
A few other notes:
* The scope of this change should probably be to remove the use of `localizeBalance` and instead add an additional `options` param to `Amount` that takes additional localization params
* Everywhere where `localizeBalance` was previously used (it's only a couple of places) should be replaced by `Amount` with the additional params

**Comment by tarunsamanta2k20** [comment]
@ckniffen are you talking about balance and balance2?

**Comment by tarunsamanta2k20** [comment]
@ckniffen is it fine?

image

**Comment by tarunsamanta2k20** [comment]
> A few other notes, after discussing offline with @ckniffen:
> 
>     * The scope of this change should probably be to remove the use of `localizeBalance` and instead add an additional `options` param to `Amount` that takes additional localization params
> 
>     * Everywhere where `localizeBalance` was previously used (it's only a couple of places) should be replaced by `Amount` with the additional params

But localizeBalance(balance, language) is a method. it is doing some operation.

**Comment by mvadari** [comment]
> > A few other notes, after discussing offline with @ckniffen:
> > ```
> > * The scope of this change should probably be to remove the use of `localizeBalance` and instead add an additional `options` param to `Amount` that takes additional localization params
> > 
> > * Everywhere where `localizeBalance` was previously used (it's only a couple of places) should be replaced by `Amount` with the additional params
> > ```
> 
> But localizeBalance(balance, language) is a method. it is doing some operation.

Yes, but what `localizeBalance` is doing is very similar to what the `Amount` element does. So `Amount` could be modified to allow for special options to be passed in, which makes `localizeBalance` unnecessary and simplifies the code.

**Comment by tarunsamanta2k20** [comment]
> > > A few other notes, after discussing offline with @ckniffen:
> > > ```
> > > * The scope of this change should probably be to remove the use of `localizeBalance` and instead add an additional `options` param to `Amount` that takes additional localization params
> > > 
> > > * Everywhere where `localizeBalance` was previously used (it's only a couple of places) should be replaced by `Amount` with the additional params
> > > ```
> > 
> > 
> > But localizeBalance(balance, language) is a method. it is doing some operation.
> 
> Yes, but what `localizeBalance` is doing is very similar to what the `Amount` element does. So `Amount` could be modified to allow for special options to be passed in, which makes `localizeBalance` unnecessary and simplifies the code.

you are saying i need to add a props to the Amount component then make it true or false where ever localized balance is used. Right?

**Comment by mvadari** [comment]
> > 
> > 
> > Yes, but what `localizeBalance` is doing is very similar to what the `Amount` element does. So `Amount` could be modified to allow for special options to be passed in, which makes `localizeBalance` unnecessary and simplifies the code.
> 
> @mvadari what is the use of this additional localization params?

What I'm suggesting is adding an additional `options` parameter to `Amount`, for options that are passed into `localizeNumber`. That's the only difference between `localizeBalance` and `Amount`,  if you take a look at what they are doing.

**Comment by tarunsamanta2k20** [comment]
> > > Yes, but what `localizeBalance` is doing is very similar to what the `Amount` element does. So `Amount` could be modified to allow for special options to be passed in, which makes `localizeBalance` unnecessary and simplifies the code.
> > 
> > 
> > @mvadari what is the use of this additional localization params?
> 
> What I'm suggesting is adding an additional `options` parameter to `Amount`, for options that are passed into `localizeNumber`. That's the only difference between `localizeBalance` and `Amount`, if you take a look at what they are doing.

@mvadari should i add that to as prop before adding to options? like that
`<Amount value={balance} displayIssuer={false} isLocalizedAmount={true}/>`

**Comment by tarunsamanta2k20** [comment]
@ckniffen @mvadari have you reviewed my recent commit?

File: src/containers/Accounts/AMM/AMMAccounts/AMMAccountHeader/AMMAccountHeader.tsx [hunk 1]
```diff
@@ -39,11 +40,17 @@ export const AMMAccountHeader = (props: { data: AmmDataType }) => {
           </div>
           <div className="values">
             <div className="title">{t('token_balance')}</div>
-            <div className="value">{localizedBalance1}</div>
+            {/* <div className="value">{localizedBalance1}</div> */}
```
Comment (mvadari) [inline src/containers/Accounts/AMM/AMMAccounts/AMMAccountHeader/AMMAccountHeader.tsx:43]: These commented lines should be removed and the variables should be deleted as well

Comment (tarunsamanta2k20) [reply src/containers/Accounts/AMM/AMMAccounts/AMMAccountHeader/AMMAccountHeader.tsx:43]: @mvadari i kept those commented lines for your reference

Comment (mvadari) [reply src/containers/Accounts/AMM/AMMAccounts/AMMAccountHeader/AMMAccountHeader.tsx:43]: You don't need to do that - it shows up in the diff

**Comment by ckniffen** [comment]
The variables `localizedBalance1` and `localizedBalance2` should be deleted now.

**Comment by ckniffen** [comment]
> Please include Before/After screenshots in the PR description at the top (you can edit the text).

Also reference the issue number in the description.  This will automatically close the issue when the PR is merged.

File: src/containers/shared/components/Amount.tsx [hunk 2]
```diff
@@ -26,16 +26,16 @@ export const Amount = ({
...
-        {modifier && <span className="amount-modifier">{modifier}</span>}
-        {localizedAmount}
-      </span>{' '}
       <Currency
         issuer={displayIssuer ? issuer : ''}
         currency={currency}
         link
         displaySymbol={false}
-      />
+      />{' '}
```
Comment (mvadari) [inline src/containers/shared/components/Amount.tsx:34]: These changes should be backed out - the `Amount` file should not need to be modified

Comment (tarunsamanta2k20) [reply src/containers/shared/components/Amount.tsx:34]: Then token code will be on the right side. is it okay?

Comment (mvadari) [reply src/containers/shared/components/Amount.tsx:34]: Yes. This change will modify a lot of other pages, and we don't want that.

**Review by ckniffen — APPROVED** [review]

**Review by mvadari — APPROVED** [review]

**Comment by ckniffen** [comment]
@tarunsamanta2k20 Thank you for this contribution.

**Pull request merged**